src/ApplicationLogs/LogReader.cs

```diff
@@ -90,14 +90,19 @@ public static JObject TxLogToJson(Blockchain.ApplicationExecuted appExec)
             trigger["vmstate"] = appExec.VMState;
             trigger["exception"] = GetExceptionMessage(appExec.Exception);
             trigger["gasconsumed"] = appExec.GasConsumed.ToString();
-            try
+            var stack = new JArray();
+            foreach (var item in appExec.Stack)
             {
-                trigger["stack"] = appExec.Stack.Select(q => q.ToJson(Settings.Default.MaxStackSize)).ToArray();
-            }
-            catch (Exception ex)
-            {
-                trigger["exception"] = ex.Message;
+                try
+                {
+                    stack.Add(item.ToJson(Settings.Default.MaxStackSize));
```

Comment:
Just a side note: it's not yet implemented in NeoGo, but how about restricting the whole list of serializable stackitems by `Settings.Default.MaxStackSize` instead of restricting every single stackitem by separate `Settings.Default.MaxStackSize`?

Reply:
> Because each StackItem can be an ExecutionEngineLimits

That's the case, we can't allow this during results serilisation, because it's not allowed for regular executions. Although it's not a problem if we use the right limits for execution: we just can't get too large number of stackitems in this case.

So resolved.